bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -583,19 +583,16 @@ public static TypedescValue getTypedesc(Object value) {
      * @param annotTag          The annot-tag-reference
      * @return the annotation value if present, nil else
      */
-    public static Object getAnnotValue(TypedescValue typedescValue, String annotTag) {
-        Type describingType = typedescValue.getDescribingType();
-        if (!(describingType instanceof BAnnotatableType)) {
-            return null;
-        }
-        return ((BAnnotatableType) describingType).getAnnotation(StringUtils.fromString(annotTag));
-    }
-
     public static Object getAnnotValue(TypedescValue typedescValue, BString annotTag) {
         Type describingType = typedescValue.getDescribingType();
         if (!(describingType instanceof BAnnotatableType)) {
             return null;
         }
+        if (describingType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG &&
+                ((BTypeReferenceType) describingType).getReferredType().getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
How come we don't have to this for records? Don't we get them as a type reference type?

Reply:
In records the annotations are added to the type reference but in the tuples they're added to the type instead. Will see if it can be added to the typeref too.

Reply:
@pcnfernando, would appreciate your thoughts too. 

Also, do we have somewhat of a deviation between what the spec defines as a type reference type and what the implementation deodeoss?

Reply:
What's the behaviour for other types?

Reply:
Eventually we'll be retrieving the annotations from the typedesc. https://github.com/ballerina-platform/ballerina-lang/pull/38488/files#diff-331f52ed2675385bed30785e051a0db1b5a477c2e14ae43521fe51a3c5519f97R602

Since the typedesc will again be of the typeref type, IMO, here again it can be on the typeref type.

Reply:
Sorry I missed this. Yes, let's have the annotation on type ref. Even for `type map<string>` it will be a type ref.
Will have to remove the special cases in https://github.com/ballerina-platform/ballerina-lang/pull/36638/files#diff-1797b4fe40dec84e2f72f5543200ab74f59207bbc477a85b86f17fa056947ad9R130

Record special case seems to be introduced with 0b722ac01118a9b6f5db166aae8a6c8420c48c80. For the moment can you create an issue mentioning the inconsistent behavior with Record types. Will have to check with Hinduja